From 4e52e7b275341b71d734ff627bd13ba7c67e52ed Mon Sep 17 00:00:00 2001
From: Trevor Davenport <trevor_davenport@selinc.com>
Date: Thu, 9 May 2024 11:51:03 -0600
Subject: [PATCH 104/105] modesetting: Fix invalid identity CTM on 32-bit.

On 32-bit, the shifts used to initialized the identity CTM overflow and
result in zero instead of the intended 1.  This results in a broken
display (on at least i915) when using the modesetting xorg driver.

Fix the invalid CTM by using ULL suffix on the shifts.

Fixes:4e670f1281ad75c5673b6ac75645036d810da8d9

Signed-off-by: Trevor Davenport <trevor_davenport@selinc.com>
Part-of: <https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/1526>
---
 hw/xfree86/drivers/modesetting/drmmode_display.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hw/xfree86/drivers/modesetting/drmmode_display.c b/hw/xfree86/drivers/modesetting/drmmode_display.c
index efcca0b03..6672cafd4 100644
--- a/hw/xfree86/drivers/modesetting/drmmode_display.c
+++ b/hw/xfree86/drivers/modesetting/drmmode_display.c
@@ -63,9 +63,9 @@ static PixmapPtr drmmode_create_pixmap_header(ScreenPtr pScreen, int width, int
                                               void *pPixData);
 
 static const struct drm_color_ctm ctm_identity = { {
-    1UL << 32, 0, 0,
-    0, 1UL << 32, 0,
-    0, 0, 1UL << 32
+    1ULL << 32, 0, 0,
+    0, 1ULL << 32, 0,
+    0, 0, 1ULL << 32
 } };
 
 static Bool ctm_is_identity(const struct drm_color_ctm *ctm)
-- 
2.45.2

